fs/expand_glob.ts

```diff
@@ -158,7 +157,9 @@ export async function* expandGlob(
     : absRoot;
   while (segments.length > 0 && !isGlob(segments[0]!)) {
     const seg = segments.shift();
-    assert(seg !== undefined);
+    if (seg === undefined) {
+      throw new ReferenceError("Unexpected undefined segment");
```

Comment:
Ditto. I think this should be `TypeError` or `Error`

Reply:
I went with `TypeError`.